### PR TITLE
Better topology serving

### LINKED ISSUE #992: Serve GeoJSON instead of TopoJSON
GeoJSON might be bigger in size, however, we might be able to save some loading time by running in-advance (precomputing) the `countries` object using `countrytopos.js` in a compile step.

### LINKED ISSUE #1046: Dash lines to mark US states within RTOs
A question for the US territories: could states within the RTOs be outlined with dash lines? This would make it easier to tell which state is located where within the RTO block. This is all aesthetics as placing the mouse of the state of Iowa versus the state of Minnesota will still display MISO as a whole regardless.

I will put forward two reasons to argue in favor of such a modification:

-When toggling the wind and/or solar potential function, it is difficult to distinguish which states are getting more wind and/or solar irradiance than others. Adding dash lines to mark state borders within RTOs would make it clear where the states' borders are within the RTO

-It could act as a (temporary) substitute for having specific, clear data of electricity generation and transport within specific state borders and at this moment, that doesn't seem to be available. The next best thing would be to simply add data for RTOs and the [unfettered boundaries](https://www.wapa.gov/newsroom/NewsFeatures/PublishingImages/ISO-RTOmap_slider.jpg) they come with. It would allow Electricitymap's representation of RTOs in the USA to be more accurate geographically while making it clear which states or which pieces of states the RTOs cover and which they do not. FOr example, it would make clear which parts of Texas are covered by SPP, MISO and ERCOT.

I would like to hear your thoughts.

### PR DESCRIPTION
**Pull request by corradio**
- ~Serves GeoJSON instead of TopoJSON~
- Precompiles the GeoJSON to avoid doing work on the client
- Supports showing state lines

Fixes #1046 
Ref #992 

The problem is that the bundle size is currently increased as world.json was ~500, and is now >1MB. This doesn't make any sense as the resulting topo should be less (we have removed unused geometries in the process?). This needs to be fixed.

@systemcatch let me know if you have any ideas.

### CODE REVIEW

**Comment by systemcatch** [comment]
To be honest I was expecting a bigger bundle size than that.  Though download size for map might be bigger the increase in performance could make up for it.  Needs testing.

**Comment by corradio** [comment]
Actually I just reverted to still serve topojson. The bundle size is still bigger than before, which shouldn't happen.

**Comment by systemcatch** [comment]
Had a scan of the code but nothing leaps out. If you put both world.json files on slack we might see a difference when viewing them.

**Comment by jarek** [comment]
@corradio check the JSON stringification - the new zonegeometries.json has a _lot_ of decimal digits ("33.52899174000001,35.055794169000066") which is not necessary at our scales (two should be more than enough) and could be increasing the file size.

**Comment by jarek** [comment]
The world.json `arcs` list also has a lot of decimal places in its coordintes (but I don't know if the old one did as well)

**Comment by maxbellec** [comment]
FYI, I only added a quantization step at the end in `generate-geometries`. I also reduced precision, from `1e5` to `1e4`, since I couldn't see the difference between both after simplification.

New size is `348KB` un-zipped, `102K` zipped

File: web/generate-geometries.js [hunk 1]
```diff
@@ -0,0 +1,568 @@
...
+  }
+});
+zoneFeaturesNoStatesMerge = toListOfFeatures(zoneFeaturesNoStatesMergeInline, false);
+
+// Write unsimplified list of geojson, with and without state merges
+fs.writeFileSync('zonegeometries.json', zoneFeatures.map(JSON.stringify).join('\n'));
+fs.writeFileSync('zonegeometriesNoStateMerge.json', zoneFeaturesNoStatesMerge.map(JSON.stringify).join('\n'));
+
+// Simplify
+// fs.writeFileSync('src/zones.json', JSON.stringify(zones));
```
Comment (corradio) [inline web/generate-geometries.js]: Remove

File: web/generate-geometries.js [hunk 2]
```diff
@@ -0,0 +1,568 @@
...
+  }
+  else{
+    zoneFeaturesNoStatesMergeInline.push(data);
+  }
+});
+zoneFeaturesNoStatesMerge = toListOfFeatures(zoneFeaturesNoStatesMergeInline, false);
+
+// Write unsimplified list of geojson, with and without state merges
+fs.writeFileSync('zonegeometries.json', zoneFeatures.map(JSON.stringify).join('\n'));
+fs.writeFileSync('zonegeometriesNoStateMerge.json', zoneFeaturesNoStatesMerge.map(JSON.stringify).join('\n'));
```
Comment (corradio) [inline web/generate-geometries.js]: Cleanup: we only need one of them

**Comment by systemcatch** [comment]
Hi @maxbellec nice to have you on board. The size of world.json is looking better and documenting the process of how it's built is a good idea.

**Comment by corradio** [comment]
Good on my side, let's merge once max has tested the frontend

**Pull request merged**